lib/TemplatedPathPlugin.js

```diff
@@ -75,130 +90,187 @@ const getReplacer = (value, allowEmpty) => {
  */
 const replacePathVariables = (path, data) => {
 	const chunkGraph = data.chunkGraph;
-	const chunk = data.chunk;
-	const chunkId = chunk && chunk.id;
-	const chunkName = chunk && (chunk.name || chunk.id);
-	const chunkHash = chunk && (chunk.renderedHash || chunk.hash);
-	const chunkHashWithLength =
-		chunk && "hashWithLength" in chunk && chunk.hashWithLength;
-	const contentHashType = data.contentHashType;
-	const contentHash =
-		data.contentHash ||
-		(chunk &&
-			contentHashType &&
-			chunk.contentHash &&
-			chunk.contentHash[contentHashType]);
-	const contentHashWithLength =
-		data.contentHashWithLength ||
-		(chunk && "contentHashWithLength" in chunk
-			? chunk.contentHashWithLength[contentHashType]
-			: undefined);
-	const module = data.module;
-	const moduleId =
-		module &&
-		(module instanceof Module ? chunkGraph.getModuleId(module) : module.id);
-	const moduleHash =
-		module &&
-		(module instanceof Module
-			? chunkGraph.getRenderedModuleHash(module)
-			: module.renderedHash || module.hash);
-	const moduleHashWithLength =
-		module && "hashWithLength" in module && module.hashWithLength;
-
-	if (typeof path === "function") {
-		path = path(data);
-	}
 
-	if (
-		data.noChunkHash &&
-		(REGEXP_CHUNKHASH_FOR_TEST.test(path) ||
-			REGEXP_CONTENTHASH_FOR_TEST.test(path))
-	) {
-		throw new Error(
-			`Cannot use [chunkhash] or [contenthash] for chunk in '${path}' (use [hash] instead)`
+	// Chunk Context
+	//
+	// Placeholders
+	//
+	// [id] - chunk.id (0.js)
+	// [ext] - TODO (app.{js, css, ...})
+	// [name] - chunk.name (app.js)
+	// [hash] - data.hash (53276435.js)
+	// [chunkhash] - chunk.hash (7823t4t4.js)
+	// [contenthash] - chunk.contentHash[type] (3256urzg.js)
+	// [query] - data.query (app.js?v=1.0.0)
+	// [file] - data.filename (/context/file.map.js)
+	// [filebase] - data.basename (file.map.js)
+	if (data.chunk) {
+		const chunk = data.chunk;
+
+		const id = chunk.id;
+		const name = chunk.name || chunk.id;
+
+		const chunkHash = chunk.renderedHash || chunk.hash;
+		// @ts-ignore
+		const chunkHashWithLength = chunk.hashWithLength || undefined;
```

Comment:
You need to do something like `"hashWithLength" in chunk` to get rid of the `@ts-ignore`.

Reply:
I changed it because `// ts-ignore` doesn't seem to work with ternary operators atm 🤷‍♂️

```js
const chunkHashWithLength = "hashWithLength" in chunk 
  ? chunk.hashWithLength // Type Linting {Error} (maybe chunk["hashWithLength"] instead ?)
  : undefined
```

I initially tried

```js
const chunkHashWithLength = "hashWithLength" in chunk 
  // ts-ignore (doesn't work :()
  ? chunk.hashWithLength // Type Linting {Error}
  : undefined
```

```js
...
  ? // ts-ignore (also fails)
    chunk.hashWithLength // Type Linting {Error}
...
```

Reply:
Ok just finish the other things and I will take care of this.